dask-gateway.yaml

```diff
@@ -38,31 +39,67 @@ pipeline:
+      # Include system site-packages
+      sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.destdir}}/usr/share/${{package.name}}/.venv/pyvenv.cfg
+
+      # Symlink scripts to PATH
+      mkdir -p ${{targets.destdir}}/usr/bin/
+      ln -s /usr/share/${{package.name}}/.venv/bin/dask ${{targets.destdir}}/usr/bin/dask
+      ln -s /usr/share/${{package.name}}/.venv/bin/dask-scheduler ${{targets.destdir}}/usr/bin/dask-scheduler
+      ln -s /usr/share/${{package.name}}/.venv/bin/dask-ssh ${{targets.destdir}}/usr/bin/dask-ssh
+      ln -s /usr/share/${{package.name}}/.venv/bin/dask-worker ${{targets.destdir}}/usr/bin/dask-worker
 
 subpackages:
   - name: dask-gateway-server
     description: A multi-tenant server for securely deploying and managing Dask clusters
+    options:
+      no-depends: true
+    dependencies:
+      runtime:
+        - python-3
     pipeline:
       - name: Python Build
         runs: |
-          cd dask-gateway-server
-          python3 -m gpep517 build-wheel --wheel-dir dist --output-fd 1
-          mkdir -p "${{targets.subpkgdir}}/usr/bin"
-          install -Dm755 ./dask_gateway_server/proxy/dask-gateway-proxy "${{targets.subpkgdir}}/usr/bin/"
-          python -m installer -d "${{targets.subpkgdir}}/" dist/dask_gateway_server*.whl
-    dependencies:
-      runtime:
-        - py3-sqlalchemy
-        - py3-typing-extensions
-        - py3-traitlets
-        - py3-colorlog
-        - py3-aiohttp
-        - py3-cryptography
-        - py3-kubernetes-asyncio
+          cd ${{package.name}}-server
+
+          # Build package
+          python -m gpep517 build-wheel --wheel-dir dist --output-fd 1
+
+          # Setup venv and install package
+          python -m venv .venv --system-site-packages
+          .venv/bin/pip install -I --no-compile dist/*.whl
+
+          # Install kubernetes asyncio, sqlalchemy, and typing extensions
+          .venv/bin/pip install kubernetes-asyncio sqlalchemy typing_extensions --no-compile
+
+          mkdir -p ${{targets.subpkgdir}}/usr/share/${{package.name}}-server
+          mv .venv ${{targets.subpkgdir}}/usr/share/${{package.name}}-server/
+
+          # Fix venv paths
+          sed -i "s|/home/build|/usr/share|g" ${{targets.subpkgdir}}/usr/share/${{package.name}}-server/.venv/bin/*
+
+          # Include system site-packages
+          sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.subpkgdir}}/usr/share/${{package.name}}-server/.venv/pyvenv.cfg
+
+          # Symlink scripts to PATH
+          mkdir -p ${{targets.subpkgdir}}/usr/bin/
+          ln -s /usr/share/${{package.name}}-server/.venv/bin/${{package.name}}-jobqueue-launcher ${{targets.subpkgdir}}/usr/bin/${{package.name}}-jobqueue-launcher
+          ln -s /usr/share/${{package.name}}-server/.venv/bin/${{package.name}}-server ${{targets.subpkgdir}}/usr/bin/${{package.name}}-server
```

Comment:
Could we look at adding some melange tests? especially given we've changed the packaging format

Reply:
Yeah. I can do that

Reply:
Added :)